### PR TITLE
Delete PASS_IF macro from FAIL/PASS API - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Link to redmine tickets:
https://redmine.openinfosecfoundation.org/issues/4795
and
https://redmine.openinfosecfoundation.org/issues/4021

Previous PR: https://github.com/OISF/suricata/pull/6546

Describe changes:
- replace `FAIL_IF` for `FAIL_IF_NOT_NULL`, and `FAIL_IF_NOT` with `FAIL_IF_NULL` calls

### CODE REVIEW

File: src/detect-dsize.c [hunk 1]
```diff
@@ -494,400 +495,277 @@ void SigParseApplyDsizeToContent(Signature *s)
...
-    if (dd) {
-        DetectDsizeFree(NULL, dd);
-        return 0;
-    }
-
-    return 1;
+    DetectDsizeData *dd = DetectDsizeParse("A");
+    FAIL_IF_NOT_NULL(dd);
+    DetectDsizeFree(NULL, dd);
+    PASS;
```
Comment (jasonish) [inline src/detect-dsize.c:612]: The free can be removed here.  If dd is not null, we'll never make it to `DetectDsizeFree` as it will have failed.  Its a bit picky, but the shorter these tests can be the easier to reason about them I think.

**Review by jasonish — COMMENTED** [review]
Just the one nit of an extra line, otherwise looks good to me.

File: src/detect-dsize.c [hunk 2]
```diff
@@ -494,400 +495,277 @@ void SigParseApplyDsizeToContent(Signature *s)
...
 }
 
 /**
  * \test this is a test for a valid dsize value 1
  *
- *  \retval 1 on succces
- *  \retval 0 on failure
  */
 static int DsizeTestParse05 (void)
```
Comment (jufajardini) [inline src/detect-dsize.c:547]: Considering we want to make unittests more compact, in general, and clean tests up when possible I think we could replace:
- `DsizeTestParse01` w/ `DsizeTestParse05` (and probably bring the check from `DsizeTestParse13` to the `05`, too)
- `DsizeTestParse02` w/ `DsizeTestParse06`
- `DsizeTestParse03` w/ `DsizeTestParse07`
- `DsizeTestParse04` w/ `DsizeTestParse08`

**Comment by suricata-qa** [comment]
WARNING:
 

ERROR: QA failed on tlpw2_autofp_alerts_cmp.
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.wrk.flows_evicted | 5309149 | 4693221 | 113.12% |
| .flow.mgr.rows_maxlen | 115 | 539 | 21.34% | 

Pipeline 4690

File: src/detect-dsize.c [hunk 3]
```diff
@@ -242,7 +242,8 @@ static DetectDsizeData *DetectDsizeParse (const char *rawstr)
         }
     }
 
-    SCLogDebug("dsize parsed successfully dsize: %"PRIu16" dsize2: %"PRIu16"",dd->dsize,dd->dsize2);
+    SCLogDebug("dsize parsed successfully dsize: %" PRIu16 " dsize2: %" PRIu16 "", dd->dsize,
+            dd->dsize2);
```
Comment (inashivb) [inline src/detect-dsize.c:246]: Is this cleanup fix unrelated to tests? Should this be a part of this commit?

Comment (inashivb) [inline src/detect-dsize.c:246]: Is this cleanup related to the patch? Should it be a part of the same commit?

Comment (jufajardini) [reply src/detect-dsize.c:246]: My bad for that. Clang sneaked in, after I used a `SCLogNotice` there. I'll undo that with next PR...

**Comment by jufajardini** [comment]
Feedback incorporated in https://github.com/OISF/suricata/pull/6570
